Bulldozer.CSV/Maps/Benevolence.cs

```diff
@@ -0,0 +1,429 @@
+
+            string[] row;
+            // Uses a look-ahead enumerator: this call will move to the next record immediately
+
+            while ( ( row = csvData.Database.FirstOrDefault() ) != null )
+            {
+                var benevolenceRequestText = row[BenevolenceRequestText];
+                var benevolenceRequestDate = row[BenevolenceRequestDate];
+                var benevolenceRequestId = row[BenevolenceRequestId];
+                var benevolenceRequestFirstName = row[BenevolenceRequestFirstName];
+                var benevolenceRequestLastName = row[BenevolenceRequestLastName];
+                var benevolenceRequestEmail = row[BenevolenceRequestEmail];
+                var benevolenceRequestCreatedById = row[BenevolenceRequestCreatedById];
+                var benevolenceRequestCreatedDate = row[BenevolenceRequestCreatedDate];
+                var benevolenceRequestRequestedById = row[BenevolenceRequestRequestedById];
+                var benevolenceRequestCaseWorkerId = row[BenevolenceRequestCaseWorkerId];
+                var benevolenceRequestCellPhone = row[BenevolenceRequestCellPhone];
+                var benevolenceRequestHomePhone = row[BenevolenceRequestHomePhone];
+                var benevolenceRequestWorkPhone = row[BenevolenceRequestWorkPhone];
+                var benevolenceRequestGovernmentId = row[BenevolenceRequestGovernmentId];
+                var benevolenceRequestProvidedNextSteps = row[BenevolenceRequestProvidedNextSteps];
+                var benevolenceRequestStatus = row[BenevolenceRequestStatus];
+                var benevolenceRequestResultSummary = row[BenevolenceRequestResultSummary];
+                var benevolenceRequestAddress = row[BenevolenceRequestAddress];
+                var benevolenceRequestAddress2 = row[BenevolenceRequestAddress2];
+                var benevolenceRequestCity = row[BenevolenceRequestCity];
+                var benevolenceRequestState = row[BenevolenceRequestState];
+                var benevolenceRequestZip = row[BenevolenceRequestZip];
+                var benevolenceRequestCountry = row[BenevolenceRequestCountry];
+
+                //
+                // Verify we have the minimum required information for a valid BenevolenceRequest in the csv file.
+                //
+                if ( string.IsNullOrWhiteSpace( benevolenceRequestText )
+                    || ( string.IsNullOrWhiteSpace( benevolenceRequestRequestedById ) && ( string.IsNullOrWhiteSpace( benevolenceRequestFirstName ) || string.IsNullOrWhiteSpace( benevolenceRequestLastName ) ) ) )
+                {
+
+                    throw new System.Collections.Generic.KeyNotFoundException( $"Benevolence Request {benevolenceRequestId} is missing information. BenevolenceRequestText and either BenevolenceRequestRequestedById or both BenevolenceRequestFirstName and BenevolenceRequestLastName are required. ", null );
```

Comment:
Does this kill the entire import after the loop? Can this scenario be tested?

Reply:
It indeed did kill the process. Just committed code to change all exception handling for the benevolence map to provide a report to the console as well as enter an exception into the exception log and keep moving.